File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let maxProfit = 0
    let minBuy = Infinity
    
    for (let p of prices) {
        minBuy = Math.min(minBuy, p)
        maxProfit = Math.max(maxProfit, p - minBuy)
    }
    return maxProfit
};

File: best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    var minPrice = math.Inf(1)
    var maxProfit = 0.0

    for _, v := range prices {
      minPrice = math.Min(minPrice, float64(v))
      maxProfit = math.Max(float64(maxProfit), (float64(v) - minPrice))
    }
    return int(maxProfit)
}